File: api/src/main/java/gg/mineral/api/entity/living/human/property/Gamemode.java
package gg.mineral.api.entity.living.human.property;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public enum Gamemode {
    SURVIVAL((byte) 0), CREATIVE((byte) 1), ADVENTURE((byte) 2), HARDCORE((byte) 0x8);

    private final byte id;
}

File: api/src/main/java/gg/mineral/api/entity/living/human/property/Gamemode.kt
package gg.mineral.api.entity.living.human.property

import lombok.AllArgsConstructor
import lombok.Getter

@AllArgsConstructor
@Getter
enum class Gamemode {
    SURVIVAL(0.toByte()), CREATIVE(1.toByte()), ADVENTURE(2.toByte()), HARDCORE(0x8.toByte());

    private val id: Byte = 0
}
